salt/engines/script.py

```diff
@@ -53,7 +63,7 @@ def _get_serializer(output):
         )
 
 
-def start(cmd, output="json", interval=1):
+def start(cmd, output="json", interval=1, onchange=False):
```

Comment:
Not positive @Ch3LL - shouldn't we default to True here to maintain backwards compatibility? If we want to change behavior in the future we can set this to None, and do `if onchanges is None: log.warning('onchanges was not set, defaulting to False but this will change after YYYY-MM-DD. To supress this warning set onchanges to True or False')`

Reply:
Not sure I'm following, but `onchange=False` exihibits the same behavior as without this PR (firing events every time the script is run). The new feature is opt-in by setting `onchange: true`.

Reply:
I agree that this does not seem to be a breaking change to me.